SnM/SnM_RegionPlaylist.cpp

```diff
@@ -1326,23 +1379,102 @@ int GetPrevValidItem(int _plId, int _itemId, bool _startWith, bool _repeat, bool
 	return -1;
 }
 
-bool SeekItem(int _plId, int _nextItemId, int _curItemId)
+void PrepareToEndPlaylist ()
+{
+	// temp override of the "stop play at project end" option
+	if (ConfigVar<int> opt = "stopprojlen") {
+		g_oldStopprojlenPref = *opt;
+		*opt = 1;
+	}
+	g_playNext = -1;
+	g_nextRegionId = -1;
+	g_rgnLoop = 0;
+	g_nextRgnPos = SNM_GetProjectLength()+1.0;
+	g_nextRgnEnd = g_nextRgnPos+1.0;
+
+	int markerIdx = 0;
+	GetLastMarkerAndCurRegion(NULL, g_curRgnEnd - SNM_FUDGE_FACTOR, &markerIdx, nullptr);
+	EnumProjectMarkers2(NULL, markerIdx, nullptr, &g_safeTimeToEndPlaylist, nullptr, nullptr, nullptr);
+}
+
+// Should only be called from SeekRegionDelayer and SeekRegion
+void SeekRegionImpl (const int _reaperRegionId)
+{
+	const double cursorpos = GetCursorPositionEx(NULL);
+	PreventUIRefresh(1);
+	GoToRegion(NULL, _reaperRegionId, false);
+	if ((GetPlayStateEx(NULL)&1) != 1)
+		OnPlayButton();
+	SetEditCurPos2(NULL, cursorpos, false, false);
+	PreventUIRefresh(-1);
+
+#if defined(_SNM_RGNPL_DEBUG1) || defined(_SNM_RGNPL_DEBUG2)
+	char dbg[128] = "";
+	snprintf(dbg, sizeof(dbg), "GoToRegion() - Reaper Region Id: %d\n", _reaperRegionId);
+	OutputDebugString(dbg);
+#endif
+}
+
+// GoToRegion timing issue workaround
+class SeekRegionDelayer {
+	static constexpr int delayAmount = 5;
```

Comment:
Maybe even a single delay would be enough, but I wouldn't go below 3, just to be safe.